craft_application/services/base.py

```diff
@@ -21,23 +21,68 @@
 
 from craft_cli import emit
 
+from craft_application import errors, models
+
 if typing.TYPE_CHECKING:
-    from craft_application import models
     from craft_application.application import AppMetadata
     from craft_application.services import ServiceFactory
 
 
 # ignoring the fact that this abstract class has no abstract methods.
 class BaseService(metaclass=abc.ABCMeta):  # noqa: B024
-    """A service containing the actual business logic of one or more commands."""
+    """Abstract base class for a service. Should not be used directly.
+
+    BaseService allows lazy-loading of a project. EagerService should be used
```

Comment:
I think there's a mix-up here: I thought the use case was having a service type that doesn't need a project at all. @sergiusens does the StoreService need to lazy-load the project? Why?

Reply:
well maybe @lengau you need a lazy-loading of the project?

Reply:
For backwards compatibility charmcraft needs to be able to load a charmcraft.yaml file (if it exists) to configure the store. I figured the StoreService would try loading the project and use the default otherwise.

However, it would be perfectly reasonable to make the service simply parse the yaml file as a dict since we don't need anything particularly complex.

Reply:
the yaml dict is a bit tricky because the service doesn't know where it is (though you can pass the working dir I suppose), and also because we're getting more sophisticated with the yaml-loading (environment expansion, build-secrets, etc). Food for weekend thought!